salt/states/linux_acl.py

```diff
@@ -221,3 +284,241 @@ def absent(name, acl_type, acl_name='', perms='', recurse=False):
+
+    Takes a list of acl names and add them to the given path
+
+    name
+        The acl path
+
+    acl_type
+        The type of the acl is used for it can be 'user' or 'group'
+
+    acl_names
+        The list of users or groups
+
+    perms
+        Set the permissions eg.: rwx
+
+    recurse
+        Set the permissions recursive in the path
+
+    force
+        Wipe out old permissions and ensure only the new permissions are set
+    '''
+    if acl_names is None:
+        acl_names = []
+    ret = {'name': name,
+           'result': True,
+           'changes': {},
+           'pchanges': {},
+           'comment': ''}
+
+    _octal = {'r': 4, 'w': 2, 'x': 1, '-': 0}
+    _octal_perms = sum([_octal.get(i, i) for i in perms])
+    if not os.path.exists(name):
+        ret['comment'] = '{0} does not exist'.format(name)
+        ret['result'] = False
+        return ret
+
+    __current_perms = __salt__['acl.getfacl'](name)
+
+    if acl_type.startswith(('d:', 'default:')):
+        _acl_type = ':'.join(acl_type.split(':')[1:])
+        _current_perms = __current_perms[name].get('defaults', {})
+        _default = True
+    else:
+        _acl_type = acl_type
+        _current_perms = __current_perms[name]
+        _default = False
+        _origin_group = _current_perms['comment']['group']
+        _origin_owner = _current_perms['comment']['owner']
+
+        _current_acl_types = list()
+        diff_perms = False
+        for key in _current_perms[acl_type]:
+            for current_acl_name in key.keys():
+                _current_acl_types.append(current_acl_name.encode('utf-8'))
+                diff_perms = _octal_perms == key[current_acl_name]['octal']
+        if acl_type == 'user':
+            _current_acl_types.remove(_origin_owner)
+        else:
+            _current_acl_types.remove(_origin_group)
+        diff_acls = set(_current_acl_types) ^ set(acl_names)
+        if not diff_acls and diff_perms and not force:
+            ret = {'name': name,
+                   'result': True,
+                   'changes': {},
+                   'pchanges': {},
+                   'comment': 'Permissions and {}s are in the desired state'.format(acl_type)}
+            return ret
+    # The getfacl execution module lists default with empty names as being
+    # applied to the user/group that owns the file, e.g.,
+    # default:group::rwx would be listed as default:group:root:rwx
+    # In this case, if acl_name is empty, we really want to search for root
+    # but still uses '' for other
+
+    # We search through the dictionary getfacl returns for the owner of the
+    # file if acl_name is empty.
+    if acl_names == '':
+        _search_name = __current_perms[name].get('comment').get(_acl_type, '')
+    else:
+        _search_name = acl_names
+
+    if _current_perms.get(_acl_type, None) or _default:
+        try:
+            user = [i for i in _current_perms[_acl_type] if next(six.iterkeys(i)) == _search_name].pop()
```

Comment:
Based on the code at lines 367 - 370 I don't see how they would ever match.  The `acl_names` would be a list and in turn _search_name would be a list, but the code in the list comprehension is going to produce a single value.